build-logic-commons/gradle-plugin/src/main/kotlin/gradlebuild.code-quality.gradle.kts

```diff
@@ -80,100 +55,100 @@ dependencies {
-                extension.enabled.filter { it }.map { dep }
-            )
-        }
-
-        // don't forget to update the version in distributions-dependencies/build.gradle.kts
-        addErrorProneDependency("com.google.errorprone:error_prone_core:2.42.0")
-        addErrorProneDependency("com.uber.nullaway:nullaway:0.12.10")
-
-        project.tasks.named<JavaCompile>(this.compileJavaTaskName) {
+        project.tasks.named<JavaCompile>(compileJavaTaskName) {
             options.errorprone {
                 isEnabled = extension.enabled
-                checks = errorproneExtension.disabledChecks.map {
-                    it.associateWith { CheckSeverity.OFF }
-                }
-
                 nullaway {
-                    checkContracts = true
-                    isJSpecifyMode = true
-                    severity = errorproneExtension.nullawayEnabled.map { if (it) CheckSeverity.ERROR else CheckSeverity.OFF }
+                    severity = errorproneExtension.nullawayEnabled.map { if (it) ERROR else OFF }
                 }
             }
         }
+        // don't forget to update the version in distributions-dependencies/build.gradle.kts
+        addErrorProneDependency(annotationProcessorConfigurationName, extension, "com.google.errorprone:error_prone_core:2.42.0")
+        addErrorProneDependency(annotationProcessorConfigurationName, extension, "com.uber.nullaway:nullaway:0.12.10")
+        addErrorProneDependency(annotationProcessorConfigurationName, extension, "tech.picnic.error-prone-support:error-prone-contrib:0.26.0")
     }
 }
 
 tasks.withType<JavaCompile>().configureEach {
     options.errorprone {
         disableWarningsInGeneratedCode = true
-        allErrorsAsWarnings = true
+        disableAllWarnings = true // considering this immense spam burden, remove this once to fix dedicated flaw. https://github.com/diffplug/spotless/pull/2766
+        disable(
+            "JavaxInjectOnAbstractMethod", // We use abstract injection as a pattern.
+            // picnic (https://error-prone.picnic.tech)
+            "ConstantNaming", // maybe patchable,
+            "DirectReturn", // We don`t want to use this. https://github.com/junit-team/junit-framework/pull/5006#discussion_r2403984446
+            "FormatStringConcatenation",
+            "IdentityConversion",
+            "LexicographicalAnnotationAttributeListing", // We don`t want to use this. https://github.com/junit-team/junit-framework/pull/5043#pullrequestreview-3330615838
+            "LexicographicalAnnotationListing", // patchable
+            "MissingTestCall",
+            "NestedOptionals",
+            "NonStaticImport", // We don`t want to use this.
+            "OptionalOrElseGet",
+            "PrimitiveComparison",
+            "StaticImport",  // patchable
+            "TimeZoneUsage",
+        )
+        error(
+            "MissingOverride",
+            // "RemoveUnusedImports",
+            "SelfAssignment",
+            // "StaticImport",
```

Comment:
should squash and activate after merge then diff will be simple.
first intro should be small.